libsixel: new package

taliaferro opened the pull request:
Libsixel is a library for dealing with sixel graphics -- a way of representing images as terminal escape sequences, which can be displayed by some terminal emulators, like iTerm2. The code, and additional information, can be found at [saitoha/libsixel](https://github.com/saitoha/libsixel).\
This package produces the `libsixel` library, and also an `img2sixel` command-line tool which can convert images to the sixel format.

I've got this to build successfully with the `+jpeg` and `+png` variants (which I set as the default) but I added variants for most of the flags supported by the `configure` script for the project.

I also didn't include the flag that also builds the Python bindings, but that could be added in the future -- either in this package or in a separate, dependent `py-libsixel` package?

taliaferro committed:
add new package libsixel

alecbcs commented on var/spack/repos/builtin/packages/libsixel/package.py:
```diff
@@ -0,0 +1,53 @@
+    depends_on("libjpeg", when="+jpeg")
+
+    variant("png", default=True, description="build with libpng")
+    depends_on("libpng", when="+png")
```
```suggestion
    variant("curl", default=False, description="build with libcurl")
    variant("gd", default=False, description="build with libgd")
    variant("gdk-pixbuf2", default=False, description="build with gdk-pixbuf2")
    variant("jpeg", default=True, description="build with libjpeg")
    variant("png", default=True, description="build with libpng")

    depends_on("curl", when="+curl")
    depends_on("gdk-pixbuf", when="+gdk-pixbuf2")
    depends_on("libgd", when="+gd")
    depends_on("libjpeg", when="+jpeg")
    depends_on("libpng", when="+png")
```

Just a quick nitpick. It makes it easier to review packages if they follow the common format of
```
1. versions
2. variants
3. dependencies
```

taliaferro replied:
I've reordered things so variants and dependencies are grouped with each other, and formatted it so it should pass the style checks. Do you have any other recommendations?

alecbcs reviewed (changes requested):
Nice job on your first package @taliaferro! Just a quick nitpick about variant/dependency formatting.

taliaferro committed:
reorder to group variants and dependencies together

taliaferro committed:
Switch to using source from fork

The original developer of libsixel, Hayaki Saito (@saitoha), disappeared
in early 2020 and has not updated the repository or been seen since.
However, a fork of the project has been created at libsixel/libsixel,
and that fork has been getting much more regular updates. In this commit
I switch the package to using the better-maintained, now apparently
canonical fork of the project.

That project switched the build system from autotools to Meson, so much
of the build arguments code needed to be rewritten. The newest official
release also contained an error in meson.build which would break on
newer versions of Meson. That error only applied to the libjpeg and
libpng variants, though, so I switched the default to use libgd which
has broader format support anyway.

taliaferro committed:
blacken

alecbcs commented on var/spack/repos/builtin/packages/libsixel/package.py:
```diff
@@ -0,0 +1,41 @@
+
+class Libsixel(MesonPackage):
+    """
+    This package provides encoder/decoder implementation for DEC SIXEL graphics, and some converter programs.
```
@taliaferro can you break up this line into to two to fix the failing style check. (The line is >100 characters long.)

taliaferro replied:
Sure, thanks for the reminder!

taliaferro committed:
broke description into multiple lines

taliaferro committed:
Merge remote-tracking branch 'upstream/develop' into add/libsixel

taliaferro committed:
allow libjpeg, etc to be loaded automatically

alecbcs reviewed (approved):
Looks good to me. Thanks @taliaferro!
